TsJournalTrello.gs

```diff
@@ -0,0 +1,361 @@
+  //If it hasnt, change the style to Heading 2 and add a Hyperlink to the Trello Card 
+  var journal = DocumentApp.openById(DocId)
+  var DocUrl = journal.getUrl()
+  var npBookmark = null
+ 
+  if (journal === null) { 
+  
+    Log_.warning('Invalid Journal ID' + DocId)
+   
+    return
+  } else {
+   
+    //Process the journal, find the next tello card task, hyperlink the heading
+    var [trelloCardTitle, npBookmarkId] = processJournal_(journal, trelloBoardId)
+          
+    if (trelloCardTitle == null) {
+      Log_.warning('Either no trello card title found or trello card found on trello board')
+      return
+    } else if (npBookmarkId == null) {
+      Log_.warning('No Bookmark Found')    
+      return
+    } else {
+      //Add the link to the timesheet
+      processTimesheet_(timesheetUrl, npBookmarkId, DocUrl, trelloCardTitle)    
+    }
+  }
+                
+  return SUCCESS  
+}// linkHeaders() 
+
+
+function processJournal_(journal, trelloBoardId) {
+          
+  var paragraphs = journal.getBody().getParagraphs()
+  var ProcessJournalReturn = null
+  var Heading1Paragraphs = []
+  
+  for(var i in paragraphs) { 
+    if (paragraphs[i].getHeading() === DocumentApp.ParagraphHeading.HEADING1) {
+      Heading1Paragraphs.push(i)
+    }
+  }
+  
+  var lastHeading1Paragraph = Heading1Paragraphs[Heading1Paragraphs.length -1]
+  var trelloCardParagraph = paragraphs[Number(lastHeading1Paragraph) + 1]
+  var trelloCardTitle = trelloCardParagraph.getText()
+  
+  if (trelloCardParagraph.getHeading() !== DocumentApp.ParagraphHeading.HEADING2) {
+    var trelloCardUrl = getTrelloCardUrl_(trelloBoardId, trelloCardTitle)
+    
+      if (trelloCardUrl == null) {
+  
+        Log_.warning('Card Name ' + trelloCardTitle + ' not found in list of Trello Boards')
+        return [null, null]
+      } else {
+  
+        var sectionPos = journal.newPosition(trelloCardParagraph, 0);
+        var npBookmark = journal.addBookmark(sectionPos)
+        
+        // Set the next paragraph to heading 2 and add the Trello Card Title URL link
+        trelloCardParagraph.setHeading(DocumentApp.ParagraphHeading.HEADING2)
+        trelloCardParagraph.setLinkUrl(trelloCardUrl)
+      
+        ProcessJournalReturn = [trelloCardTitle, npBookmark.getId()]
+        return ProcessJournalReturn 
+     }
+  } else {
+    Log_.warning('No new trello card found')
+    return [null, null]
+  }
+       
+}
+          
+function processTimesheet_(timesheetUrl, npBookmarkId, DocUrl, trelloCardTitle) {
+  
+  if (timesheetUrl === '') {
+
+    Log_.warning('No Timesheet URL found in Org Spreadsheet')
+    return
+    
+  } else {
+    //Get the last row with data in the Date column, use this to add the Trello Card and 
+    //Journal URL link to the timesheet
+    var timeSheet = SpreadsheetApp.openByUrl(timesheetUrl).getSheetByName('Timesheet')
+    var tsDate = timeSheet.getRange(DATE_COL_).getValues()
+    var lastTsRow = tsDate.filter(String).length 
```

Comment:
That's interesting. How does that work?

Reply:
It takes the date column, and filters it on all the cells with a string in, and returns the lenght of this string, this is so I can get the last row of the timesheet.